src/component/feature/FeatureView/FeatureVariants/FeatureVariantsList/AddFeatureVariant/AddFeatureVariant.tsx

```diff
@@ -360,40 +367,40 @@ const AddVariant = ({
                             id="variant-payload-type"
                             name="type"
                             label="Type"
-                            className={commonStyles.fullWidth}
+                            className={styles.select}
                             value={payload.type}
                             options={payloadOptions}
                             onChange={onPayload}
-                            style={{ minWidth: '100px', width: '100%' }}
                         />
                     </Grid>
                     <Grid item md={8} sm={8} xs={6}>
                         <TextField
-                            rows={1}
-                            label="Value"
+                            rows={payload.type === 'json' ? 10 : 0}
```

Comment:
I think you could use the Input component here. Then you will get the error placement and styling for free.